src/helpers/delete-address-from-server.ts

```diff
@@ -0,0 +1,68 @@
+import { getUserInfo } from '../app/api';
+import { removeAddressById } from '../app/api/remove-address';
+import { getAuthToken } from '../app/ecommerce/get-auth-token';
+import { defaultBillingState } from '../app/state/profile/default-billing-state';
+import { defaultShippingState } from '../app/state/profile/default-shipping-state';
+import { optionalBillingState } from '../app/state/profile/optional-billing-state';
+import { optionalShippingState } from '../app/state/profile/optional-shipping-state';
+import type { FetchError } from '../app/types';
+import { MESSAGES, SERVER_ERROR_MESSAGES } from '../components/profile/constants';
+import { createPopupMessage } from '../shared/components/popup';
+import { resetStateToDefault } from './reset-state-to-default';
+
+export async function deleteAddressFromServer(
+  type: 'shipping' | 'billing' | 'optional-shipping' | 'optional-billing',
+): Promise<FetchError | void> {
+  try {
+    //! Delete in the future when I save token via local/session storage
+    const token = await getAuthToken('ivanIvanov@yandex.ru', 'Ivan12345');
+
+    if (typeof token !== 'string') {
+      return { message: 'Failed to get token to update Personal Data' };
+    }
+
+    const user = await getUserInfo(token);
+
+    if (!('id' in user)) {
+      return { message: 'Failed to get User Data' };
+    }
+
+    const optionalShippingAddressId = user.shippingAddressIds?.find(
+      (address) => address !== user.defaultShippingAddressId,
+    );
+    const optionalBillingAddressId = user.billingAddressIds?.find(
+      (address) => address !== user.defaultBillingAddressId,
+    );
+
+    console.log(user.shippingAddressIds, 'и', user.billingAddressIds);
+    console.log(optionalShippingAddressId);
+    console.log(optionalBillingAddressId);
+
+    const addressID =
+      type === 'shipping'
+        ? user.defaultShippingAddressId
+        : type === 'billing'
+          ? user.defaultBillingAddressId
+          : type === 'optional-shipping'
+            ? optionalShippingAddressId
+            : optionalBillingAddressId;
+
+    const state =
+      type === 'shipping'
+        ? defaultShippingState
+        : type === 'billing'
+          ? defaultBillingState
+          : type === 'optional-shipping'
+            ? optionalShippingState
+            : optionalBillingState;
```

Comment:
it's possible to map those types

```suggestion
type AddressInfo = {
  id: string | undefined;
  state: ProfileDataState;
};

function getAddress(type: "billing" | "shipping" | "optional-shipping" | "optional-billing", user: Customer): AddressInfo {
  const types = {
  shipping: {
    id: user.defaultShippingAddressId,
    state: defaultShippingState,
  },
  billing: {
    id: user.defaultBillingAddressId,
    state: defaultBillingState,
  },
  'optional-shipping': {
    id: user.shippingAddressIds?.find((id) => id !== user.defaultShippingAddressId),
    state: optionalShippingState,
  },
  'optional-billing': {
    id: user.billingAddressIds?.find((id) => id !== user.defaultBillingAddressId),
    state: optionalBillingState,
  },
  }
  return types[type];
}

    const { id: addressID, state } = getAddress(type, user);

```

The code is a little bit impromptu and might require additional fixing but it works for me